packages/app/src/systems/Transaction/hooks/useTxs.tsx

```diff
@@ -0,0 +1,114 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { arrayify, ReceiptCoder, ReceiptType, TransactionCoder } from 'fuels';
+import type { TransactionResultReceipt, Address, BN } from 'fuels';
+import { useMemo } from 'react';
+
+import { parseTx } from '../utils';
+
+import type {
+  AddressTransactionsQuery,
+  ReceiptFragment,
+} from './__generated__/operations';
+import { useAddressTransactionsQuery } from './__generated__/operations';
+
+import { useChainInfo } from '~/systems/Network';
+
+/** @TODO: Move this logic to the SDK */
+const processGqlReceipt = (
+  gqlReceipt: ReceiptFragment
+): TransactionResultReceipt => {
+  const receipt = new ReceiptCoder().decode(
+    arrayify(gqlReceipt.rawPayload),
+    0
+  )[0];
+
+  switch (receipt.type) {
```

Comment:
This should not be a simple `if (ReceiptType.ReturnData || ReceiptType.LogData)` instead of a switch? 🤔